### ISSUE TITLE
DetailPanel expand / collapse animation?

### ISSUE THREAD

**Issue by appukuttan-shailesh** [material-table-core/core#252]
Bringing up a [previously requested feature](https://github.com/mbrn/material-table/issues/1566) which never got a resolution:

Would it be possible to have expand/collapse animation when the detail panel is toggled?
Currently is simply appears when toggled on and disappears when toggled off.... doesn't look great.

I have tried using:
`import Collapse from '@material-ui/core/Collapse';`
and wrapping it around the detailpanel output. But it doesn't affect the behavior.

Would this be possible? Any tips?

**Comment by matthewoestreich** [comment]
I like this idea.  I do not believe it is currently possible, but would def be open to adding this feature!

We are actively deploying here, so we are also accepting PRs... If you (or anyone else that you may know) wants to send over a PR to resolve older issues we will gladly merge them ASAP.

Unfortunately, since this isn't really a bug and is adding a feature, it will have to be lower on my list of things to address. With that being said, and as mentioned above, we are accepting PRs if you don't feel like waiting on me lol.

Thanks again!!

**Comment by appukuttan-shailesh** [comment]
Hi @oze4 ... I am actually keen to give it a try.
That said, I have to say that even though I am quite well versed with JS/ReactJS, I am not very experienced woking on npm packages. I wanted to try to make some changes to the fork and test things out. I tried changing `package.json` to:
>"@material-table/core": "appukuttan-shailesh/core#branch_2_3_39_mod",
corresponding to: https://github.com/appukuttan-shailesh/core/tree/branch_2_3_39_mod

But when I delete my existing 'node_modules' and try to install all packages (with above updated package.json), I get the error:
> Module not found: Can't resolve '@material-table/core' in '/home/shailesh/gits/projname/subdir/src'

I am obviously doing something wrong. Any tips? Are there any guidelines anywhere that I could follow to get over the npm packaging bits. Thanks.

**Comment by matthewoestreich** [comment]
No worries!

A quick way to see what a package lets you do with it is to check out the "scripts" section in the package.json file. If you wanted to test changes to your own fork you can do:

- Fork the repo (you seem to have already done this)
- Make the changes you want to test
- Run `npm start`
  - This builds and serves a "demo" app
  - [The demo app is located here](https://github.com/material-table-core/core/tree/master/__tests__/demo)
  - You can add your own demos here to test the changes you made
  - `npm start` lets you see those changes in a browser

You shouldn't have to publish your own package, etc.. if you wanted to quickly test some changes locally.

Please let me know if you have any questions!

**Comment by matthewoestreich** [comment]
If the changes you make work, and are good, then you would create a Pull Request using your fork. From there, we would merge your fork into the main branch and then publish to NPM.

In my opinion, you should use the local [GitHub App](https://desktop.github.com/) in order to familiarize yourself with Git/GitHub. It is basically a GUI for Git and will help you learn the basics (I'm not sure how familiar with Git you are).

**Comment by appukuttan-shailesh** [comment]
Thanks @oze4 for the pointers. I am actually fine on the GitHub and ReactJS front. What I very new to is npm packaging and installations.

The part I am stuck with is:
> Make the changes you want to test

Instead of using the demo app (included within material-table-core), I was trying to install material-table-core from my GitHub fork to my current working app.... and I am failing in getting this to work.

I believe I should make changes to the source file (probably 'material-table-core/src/components/m-table-body-row.js' in this case.... still figuring it out). So I [forked](https://github.com/appukuttan-shailesh/core) the GitHub repo. For starters, I should be able to run my existing app by installing 'material-table-core' from my forked repo.

For example, I have a a completely working app that makes use of material-table-core. The package.json is something like this:

```
{
  "name": "project_name",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@material-table/core": "^3.0.16",
    "@material-ui/core": "^4.11.4",
    "@material-ui/icons": "^4.11.2",
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "axios": "^0.21.1",
    "notistack": "^1.0.8",
    "nunjucks": "^3.2.3",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-lines-ellipsis": "^0.15.0",
    "react-scripts": "4.0.3",
    "react-slick": "^0.28.1",
    "react-twitter-embed": "^3.0.3",
    "salted-md5": "^4.0.2",
    "showdown": "^1.9.1",
    "showdown-katex": "^0.8.0",
    "slick-carousel": "^1.8.1",
    "web-vitals": "^1.0.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```
As you would see, my app uses "3.0.16" and currently works fine.
But now I when I want it to install from my github repo, I change above 'package.json' to 
>"@material-table/core": "github:appukuttan-shailesh/core",

and also delete the existing package-lock.json

But when I do this the app no longer runs on doing `npm start` and I see the error:
> ./src/App.js    
Module not found: Can't resolve '@material-table/core' in 
'/home/shailesh/gits/projname/subdir/src'

If I could somehow learn how to work with local copy of my package or install from my github repo, I would be able to make some attempt on implementing this feature. The JS bits itself do not seem so daunting, but I am totally new to npm packaging bits.

**Comment by appukuttan-shailesh** [comment]
p.s. I did now manage to get the built-in demo app running, so that helps with working on this feature request. Thanks @oze4 .
But I am still curious what I was doing wrong when trying to install this package from GitHub for my existing project.

**Comment by matthewoestreich** [comment]
To use in your existing project you would have had to do something like this:

Folder Structure:

- /package.json
- /src/
  - /...other stuff.../
  - /libs/
    - /MaterialTable/  <<--- create this in your current project

Easiest way would be to just copy your local fork into the MaterialTable folder then import it like:

```js
// obviously this is relative
// lets say this file is in src/thisfile.js
import MaterialTable from './libs/MaterialTable';
```

**The important thing to note here, regardless of folder structure, is that you will need to copy your local fork into your project, and them import MaterialTable via "dot sourcing"** (`import x from '../../../x'; // <- "dot sourcing"`)

**Comment by matthewoestreich** [comment]
@appukuttan-shailesh see response above

**Comment by appukuttan-shailesh** [comment]
Ok, that explains a lot! Thanks @oze4 for the detailed response. I will try this next time.
